### ISSUE TITLE
Accessibility for asterisk required label on forms

### ISSUE THREAD

**Issue by jamesmacwhite** [verbb/formie#1177]
### What are you trying to do?

When a field is set to required, this outputs a span label with the following:

```
<span class="fui-required">*</span>
```

This may or may not be ignored by screen readers, given the contents is a special character. As the related input has aria-required set, this should announce "[Label name required]" correctly, therefore the asterisk label could be potentially hidden from screen readers to avoid confusing context, possibly with `aria-hidden` to avoid it being potentially read out, as it is more of a visual indicator with no direct meaning.

Alternatively, visually hidden labelling could be applied, but in theory the ARIA properties on the input fields relative to the label should provide this.

What are you thoughts?

### What's your proposed solution?

Add `aria-hidden` to asterisk span to avoid context issues for screen readers.

**Comment by engram-design** [comment]
Probably not a bad idea, updated for the next release. To get this early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-4 as 2.0.17",
  "...": "..."
}
```

Then run `composer update`.

**Comment by jamesmacwhite** [comment]
Awesome. Another pattern is to have a visually hidden label for the asterisk character alongside, but given ARIA is being used on the fields accordingly, I'm not sure that's needed.

There's a few interpretations I've read which differ, although in this case the main thing would be to avoid a confusing context like "First name asterisk" being announced, but it depends on screen readers really. So might just ignore it anyway, but can't hurt to make sure it's not even announced I guess.

**Comment by engram-design** [comment]
Fixed in [2.0.18](https://github.com/verbb/formie/releases/tag/2.0.18)
